server/commands/write_authzmodel.go

```diff
@@ -60,7 +65,13 @@ func (w *WriteAuthorizationModelCommand) Execute(ctx context.Context, req *openf
 //  2. For every rewrite the relations in the rewrite must:
 //     a. Be valid relations on the same type in the authorization model (in cases of computedUserset)
 //     b. Be valid relations on another existing type (in cases of tupleToUserset)
-func validateAuthorizationModel(tds []*openfgapb.TypeDefinition) error {
+//
+// If it is a 1.1 model (with types on relations), additionally:
+//  3. Every type on a relation must be a valid type:
+//     a. For a type (e.g. user) this means checking that this type is in the model
+//     b. For a type#relation this means checking that this type with this relation is in the model
+//  4. Check that a relation is assignable if and only if it has a non-zero list of types
+func validateAuthorizationModel(schemaVersion typesystem.SchemaVersion, tds []*openfgapb.TypeDefinition) error {
```

Comment:
Let's put this and similar methods under the `typesystem` package since it is logic specific to the validation of types in an authorization model.

Reply:
I love this idea, but do you mind if I do it in a following PR? I have one PR to follow this which will introduce the `typesystem` package and get the schema version into the datastore and that has already blown up like crazy.

Reply:
I'd prefer we do it here since it is technically part of the scope of this work. This PR introduces the `typesystem` package and since this method is a static method, there shouldn't be any reason why we can't just quickly copy/paste it to that package and then reference it. We're already importing the `typesystem` package in this code, so we may as well move this to where it belongs now. I don't see a need to postpone that for another PR.

Reply:
Also, do you mind changing this to take an `AuthorizationModel` instead of type definitions?

Reply:
Alright. Let's do it 🚀